server/services/conditionService.ts

```diff
@@ -136,10 +136,17 @@ export default class ConditionService {
     const conditionsWithUploads = conditions.filter(
       (condition: AdditionalCondition) => condition?.uploadSummary?.length > 0
     )
-    const additionalConditions = conditions.filter(
-      (c: AdditionalCondition) => !conditionsWithUploads.find((c2: AdditionalCondition) => c.id === c2.id)
-    )
-    return { conditionsWithUploads, additionalConditions }
+    const conditionsByCode = conditions
+      // This filter will need to be removed when the frontend handles conditions with uploads transparantly
+      .filter((c: AdditionalCondition) => {
+        return !conditionsWithUploads.find((c2: AdditionalCondition) => c.id === c2.id)
+      })
+      .reduce((acc, c) => {
+        acc[c.code] = [...(acc[c.code] || []), c]
```

Comment:
How does `[...(acc[c.code] || []), c]` work?
My initial interpretation would be that it will always set the first item in the array to be an empty array itself, ie the accumulator will end up with:
```{"abc123": [ [], <condition1>, <condition2> ] }```
That doesn't seem like it would be correct though!

Reply:
`x = (acc[c.code] || [])`

if this is the first time we've seen this condition then `x = []` otherwise `x` is the previously seen conditions in an array.

`[...x, c]` all the previous conditions and this condition added to the end. 

So basically just makes sure there is an array, and then keeps adding to the end of the array.

I'm going to extract out a `chunkBy` function into utils in a follow up so I'll make this a bit easier to read then

Reply:
Ah I misread `...(acc[c.code] || [])` as `(...acc[c.code] || [])`, having the spread operator outside of the parentheses makes a lot more sense!